From 638ca89c53e2b897cfb3f627f4acbc7d09af2f4c Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Tue, 10 Jun 2014 15:51:15 +0200
Subject: [PATCH] udev: really exclude device-mapper from block device
 ownership event locking

Arguments were wrong order, no?
This fixes commits:

e918a1b5a94f270186dca59156354acd2a596494
3d06f4183470d42361303086ed9dedd29c0ffc1b
---
* dreisner: avoid second hunk of patch which doesn't apply to v213

 src/udev/udevd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index 0f3f3f0..160360e 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -304,7 +304,7 @@ static void worker_new(struct event *event)
                          * <kabi_> IMHO this sounds like a good plan for this moment
                          */
                         if (streq_ptr("block", udev_device_get_subsystem(dev)) &&
-                            !startswith("dm-", udev_device_get_sysname(dev))) {
+                            !startswith(udev_device_get_sysname(dev), "dm-")) {
                                 struct udev_device *d = dev;
 
                                 if (streq_ptr("partition", udev_device_get_devtype(d)))
-- 
2.0.0

